dm/syncer/data_validator.go

```diff
@@ -138,52 +224,167 @@ func (v *DataValidator) fillResult(err error, needLock bool) {
 		defer v.Unlock()
 	}
 
-	var errs []*pb.ProcessError
-	if utils.IsContextCanceledError(err) {
-		v.L.Info("filter out context cancelled error", log.ShortError(err))
-	} else {
-		errs = append(errs, unit.NewProcessError(err))
-	}
-
+	// todo: if error, validation is stopped( and cancelled), and we may receive a cancelled error.
+	// todo: do we need this cancel field?
```

Comment:
For other units like syncer, this field can tell paused-by-user from paused-by-error.